# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/rustc/hotfix-sparc-nix.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

error[E0425]: cannot find value `SIGSTKFLT` in crate `libc`
   --> rustc-1.86.0-src/vendor/nix-0.29.0/src/sys/signal.rs:76:9
    |
76  |         SIGSTKFLT,A

--- rustc-1.86.0-src/vendor/nix-0.29.0/src/sys/signal.rs.vanilla	2025-04-13 13:54:43.424057915 +0200
+++ rustc-1.86.0-src/vendor/nix-0.29.0/src/sys/signal.rs	2025-04-13 13:57:54.227045403 +0200
@@ -72,6 +72,8 @@
                           target_arch = "mips32r6",
                           target_arch = "mips64",
                           target_arch = "mips64r6",
+                          target_arch = "mips",
+                          target_arch = "sparc",
                           target_arch = "sparc64"))))]
         SIGSTKFLT,
         /// To parent on child stop or exit
@@ -152,6 +154,7 @@
                     target_arch = "mips32r6",
                     target_arch = "mips64",
                     target_arch = "mips64r6",
+                    target_arch = "sparc",
                     target_arch = "sparc64"
                 ))
             ))]
@@ -234,6 +237,7 @@
                     target_arch = "mips32r6",
                     target_arch = "mips64",
                     target_arch = "mips64r6",
+                    target_arch = "sparc",
                     target_arch = "sparc64"
                 ))
             ))]
@@ -321,6 +325,7 @@
         target_arch = "mips32r6",
         target_arch = "mips64",
         target_arch = "mips64r6",
+        target_arch = "sparc",
         target_arch = "sparc64"
     ))
 ))]
@@ -338,6 +343,7 @@
         target_arch = "mips32r6",
         target_arch = "mips64",
         target_arch = "mips64r6",
+        target_arch = "sparc",
         target_arch = "sparc64"
     )
 ))]
